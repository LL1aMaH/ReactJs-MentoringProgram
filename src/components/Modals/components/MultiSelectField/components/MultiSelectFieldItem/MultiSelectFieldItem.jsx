import React, { memo, useCallback } from 'react';
import styles from '../../../../../../../styles/MultiSelectFieldItem.module.scss';

import { CheckboxField } from '../CheckboxField/CheckboxField';

export const MultiSelectFieldItem = memo(function MultiSelectFieldItem(
  props,
) {
  const { value, label, checked, disabled = false, option, onChange, onMouseEnter } = props;

  const handleChange = useCallback(() => {
    onChange(option);
  }, [onChange, option]);

  const handleMouseEnter = useCallback(() => {
    onMouseEnter(option);
  }, [onMouseEnter, option]);

  return (
    <li className={styles.option} onMouseEnter={handleMouseEnter}>
      <CheckboxField
        id={value}
        label={label}
        checked={checked}
        disabled={disabled}
        onChange={handleChange}
        className={styles.checkbox}
      />
    </li>
  );
});
