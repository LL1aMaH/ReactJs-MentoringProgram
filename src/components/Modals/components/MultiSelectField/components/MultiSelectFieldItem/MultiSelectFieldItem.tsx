import React, { memo, useCallback } from 'react';
import { CheckboxField } from '../CheckboxField';
import styles from './MultiSelectFieldItem.css';

export type TMultiSelectFieldItemProps = {
  value: string;
  label: string;
  option: string;
  checked: boolean;
  disabled?: boolean;
  onChange: (option: string) => void;
  onMouseEnter: (value: string) => void;
};

export const MultiSelectFieldItem = memo(function MultiSelectFieldItem(
  props: TMultiSelectFieldItemProps,
): JSX.Element {
  const { value, label, checked, disabled = false, option, onChange, onMouseEnter } = props;

  const handleChange = useCallback(() => {
    onChange(option);
  }, [onChange, option]);

  const handleMouseEnter = useCallback(() => {
    onMouseEnter(option);
  }, [onMouseEnter, option]);

  return (
    <li className={styles.option} onMouseEnter={handleMouseEnter}>
      <CheckboxField
        id={value}
        label={label}
        checked={checked}
        disabled={disabled}
        onChange={handleChange}
        className={styles.checkbox}
      />
    </li>
  );
});
