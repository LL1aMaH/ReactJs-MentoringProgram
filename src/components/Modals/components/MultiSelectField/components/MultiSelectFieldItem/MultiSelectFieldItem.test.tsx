import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { MultiSelectFieldItem } from './MultiSelectFieldItem';

describe('MultiSelectFieldItem', () => {
  const getComponent = (props = {}) =>
    render(
      <MultiSelectFieldItem
        value="test"
        label="label"
        checked
        option="test"
        onChange={jest.fn()}
        onMouseEnter={jest.fn()}
        {...props}
      />,
    );

  it('should render correctly', () => {
    expect(getComponent().asFragment()).toMatchSnapshot();
  });

  it('should call onMouseEnter', () => {
    const testOnMouseEnter = jest.fn();
    const { container } = getComponent({
      onMouseEnter: testOnMouseEnter,
    });

    fireEvent.mouseEnter(container.querySelector('li') as HTMLElement);

    expect(testOnMouseEnter).toBeCalledWith('test');
  });
});
