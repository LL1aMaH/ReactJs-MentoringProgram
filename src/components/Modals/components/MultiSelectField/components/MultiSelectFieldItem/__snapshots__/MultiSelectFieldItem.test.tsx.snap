// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectFieldItem should render correctly 1`] = `
<DocumentFragment>
  <li
    class="option"
  >
    <div
      class="container checkbox"
    >
      <input
        checked=""
        class="checkboxField"
        data-testid="checkbox"
        id="test"
        type="checkbox"
        value=""
      />
      <span
        class="checkMark"
      />
      <label
        class="label"
        for="test"
      >
        label
      </label>
    </div>
  </li>
</DocumentFragment>
`;
